server/src/test/java/org/elasticsearch/cluster/routing/WriteLoadPerShardSimulatorTests.java

```diff
@@ -0,0 +1,204 @@
+import static org.hamcrest.Matchers.lessThan;
+import static org.hamcrest.Matchers.sameInstance;
+
+public class WriteLoadPerShardSimulatorTests extends ESTestCase {
+
+    private static final RoutingChangesObserver NOOP = new RoutingChangesObserver() {
+    };
+    public static final String[] INDICES = { "indexOne", "indexTwo", "indexThree" };
+
+    /**
+     * We should not adjust the values if there's no movement
+     */
+    public void testNoShardMovement() {
+        final var originalNode0WriteLoadStats = randomUsageStats();
+        final var originalNode1WriteLoadStats = randomUsageStats();
+        final var allocation = createRoutingAllocation(originalNode0WriteLoadStats, originalNode1WriteLoadStats, Set.of());
+
+        final var writeLoadPerShardSimulator = new WriteLoadPerShardSimulator(allocation);
+        final var calculatedNodeUsageStates = writeLoadPerShardSimulator.nodeUsageStatsForThreadPools();
+        assertThat(calculatedNodeUsageStates, Matchers.aMapWithSize(2));
+        assertThat(
+            calculatedNodeUsageStates.get("node_0").threadPoolUsageStatsMap().get("write"),
+            sameInstance(originalNode0WriteLoadStats)
+        );
+        assertThat(
+            calculatedNodeUsageStates.get("node_1").threadPoolUsageStatsMap().get("write"),
+            sameInstance(originalNode1WriteLoadStats)
+        );
+    }
+
+    public void testMovementOfAShardWillReduceThreadPoolUtilisation() {
+        final var originalNode0WriteLoadStats = randomUsageStats();
+        final var originalNode1WriteLoadStats = randomUsageStats();
+        final var allocation = createRoutingAllocation(originalNode0WriteLoadStats, originalNode1WriteLoadStats, Set.of());
+        final var writeLoadPerShardSimulator = new WriteLoadPerShardSimulator(allocation);
+
+        // Relocate a random shard from node_0 to node_1
+        final var randomShard = randomFrom(StreamSupport.stream(allocation.routingNodes().node("node_0").spliterator(), false).toList());
+        final var moveShardTuple = allocation.routingNodes().relocateShard(randomShard, "node_1", randomNonNegativeLong(), "testing", NOOP);
+        writeLoadPerShardSimulator.simulateShardStarted(moveShardTuple.v2());
+
+        final var calculatedNodeUsageStates = writeLoadPerShardSimulator.nodeUsageStatsForThreadPools();
+        assertThat(calculatedNodeUsageStates, Matchers.aMapWithSize(2));
+
+        // Some node_0 utilization should have been moved to node_1
+        assertThat(
+            getAverageWritePoolUtilization(writeLoadPerShardSimulator, "node_0"),
+            lessThan(originalNode0WriteLoadStats.averageThreadPoolUtilization())
+        );
+        assertThat(
+            getAverageWritePoolUtilization(writeLoadPerShardSimulator, "node_1"),
+            greaterThan(originalNode1WriteLoadStats.averageThreadPoolUtilization())
+        );
+    }
+
+    public void testMovementFollowedByMovementBackWillNotChangeAnything() {
```

Comment:
This test is essentially identical to `testMovementOfAShardWillReduceThreadPoolUtilisation`, except some additional work at the end to move the shard back. It seems redundant? Maybe delete the shorter one, and rename the remaining test?

Reply:
Consolidated in f60029fcfc565d7e94b4e77400accd1d4677484c